src/rdkafka_proto.h

```diff
@@ -565,6 +567,73 @@ typedef struct rd_kafka_buf_s rd_kafka_buf_t;
         (8 + 4 + 4 + 1 + 4 + 2 + 4 + 8 + 8 + 8 + 2 + 4)
 
 
+/**
+ * @brief UUID
+ *
+ * @param most_significant_bits - most significant 64 bits for the UUID
+ * @param least_significant_bits - least significant 64 bits for the UUID
+ * @param base64str - base64 encoding for the uuid. By default, it is lazy
+ * loaded. Use function `rd_kafka_uuid_base64str()` as a getter for this field.
+ */
+typedef struct rd_kafka_uuid_s {
+        int64_t most_significant_bits;
+        int64_t least_significant_bits;
+        char base64str[23];
+} rd_kafka_uuid_t;
+
+#define RD_KAFKA_ZERO_UUID                                                     \
+        { 0, 0, "" }
+
+#define RD_KAFKA_METADATA_TOPIC_ID                                             \
+        { 0, 1, "" }
+
+/**
+ * Initialize given UUID to zero UUID.
+ *
+ * @param uuid - UUID to initialize.
+ */
+static RD_INLINE RD_UNUSED void rd_kafka_uuid_init(rd_kafka_uuid_t *uuid) {
```

Comment:
I think this isn't needed, at least at the moment, we can use `RD_ZERO_INIT` or `rd_calloc` when allocating on the heap.

Reply:
I am trying to create main initialisation and destroy functions. Planning to create a `new()` function as well with `rd_calloc()`.